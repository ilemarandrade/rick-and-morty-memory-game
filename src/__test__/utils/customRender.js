import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CharactersProvider from "../../providers/CharactersProvider";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <CharactersProvider>{children}</CharactersProvider>
    </BrowserRouter>
  );
};

const customRender = (ui, { ...options } = {}) =>
  render(ui, {
    wrapper: ({ children }) => <AllTheProviders children={children} />,
    ...options,
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
