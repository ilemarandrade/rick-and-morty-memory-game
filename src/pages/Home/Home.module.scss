@use "../../App.scss";

.containerButton {
  @extend %perfect-centered;
  position: fixed;
  bottom: 10px;
  left: 0px;
  width: 100%;
}
