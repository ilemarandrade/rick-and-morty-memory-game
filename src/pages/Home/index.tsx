import { NavLink } from "react-router-dom";
import Button from "../../components/Button";
import GameBoard from "../../components/GameBoard";
import routes from "../../constants/routes";
import { useGameControlState } from "../../contexts/GameControlContext";

const Home = () => {
  const { characters } = useGameControlState();

  return (
    <>
      <h2 className="mb-1-5">Personajes</h2>
      {characters.length && <GameBoard cardsData={characters} />}
      <div className="perfect-centered">
        <Button label="Jugar" component={NavLink} to={routes.PLAY} />
      </div>
    </>
  );
};

export default Home;
