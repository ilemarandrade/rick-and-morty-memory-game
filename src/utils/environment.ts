const env = {
  API_BASE_URL: process.env.REACT_APP_API_URL,
  APP_ENV: process.env.REACT_APP_APP_ENV,
};

export default env;
