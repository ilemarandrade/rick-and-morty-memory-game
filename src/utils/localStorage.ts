export const KEY_SHOW_INTRUCTIONS = "show-intructions-key";

export const getKey = (key: string) => localStorage.getItem(key);

export const setKey = ({ key, value }: { key: string; value: string }) =>
  localStorage.setItem(key, value);
