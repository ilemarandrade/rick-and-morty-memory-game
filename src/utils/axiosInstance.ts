import axios from "axios";
import environment from "./environment";

const instance = axios.create({
  baseURL: environment.API_BASE_URL,
  headers: {
    Accept: "application/json",
    "Content-Type": "application/json",
  },
});

export default instance;
