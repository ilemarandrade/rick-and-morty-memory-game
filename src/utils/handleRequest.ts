import { AxiosRequestConfig, Method } from "axios";
import axiosInstance from "./axiosInstance";
interface IhandleRequest<T> {
  method: Method;
  url: string;
  config?: AxiosRequestConfig;
  onSuccess: (res: { data: T }) => void;
  onError: (res: { data: T }) => void;
}

const handleRequest = <T>({
  method,
  url,
  config,
  onSuccess,
  onError,
}: IhandleRequest<T>) => {
  axiosInstance({
    method,
    url,
    ...config,
  })
    .then((res) => {
      onSuccess(res);
    })
    .catch((err) => {
      onError(err);
    });
};

export default handleRequest;
