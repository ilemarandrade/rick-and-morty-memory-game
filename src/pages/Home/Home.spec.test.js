import { render, screen, waitFor } from "../../__test__/utils/customRender";
import Home from "./index";

describe("Home", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it("should render without errors", async () => {
    const { container } = render(<Home />);
    // snapshot testing
    await waitFor(() => {
      screen.getAllByText("Rick Sanchez");
    });
    expect(container).toMatchSnapshot();
  });
});
