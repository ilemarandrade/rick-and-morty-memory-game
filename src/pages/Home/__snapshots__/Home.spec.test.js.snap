// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render without errors 1`] = `
<div>
  <h2
    class="mb-1-5"
  >
    Personajes
  </h2>
  <div
    class="root"
  >
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
    <button
      class="
      card-open card "
    >
      <img
        alt="Rick Sanchez"
        src="https://placekitten.com/200/200"
      />
      <p
        class="name"
        title="Rick Sanchez"
      >
        Rick Sanchez
      </p>
      <p
        class="origin"
      >
        Alive - Human
      </p>
    </button>
  </div>
  <div
    class="containerButton"
  >
    <a
      class="button-base button-color-primary"
      color="primary"
      href="/play"
    >
      Jugar
    </a>
  </div>
</div>
`;
